api4/status.go

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package api4
+
+import (
+	"net/http"
+
+	l4g "github.com/alecthomas/log4go"
+
+	"github.com/mattermost/platform/app"
+	"github.com/mattermost/platform/model"
+	"github.com/mattermost/platform/utils"
+)
+
+func InitStatus() {
+	l4g.Debug(utils.T("api.status.init.debug"))
+
+	BaseRoutes.User.Handle("/status", ApiHandler(getUserStatus)).Methods("GET")
+
+}
+
+func getUserStatus(c *Context, w http.ResponseWriter, r *http.Request) {
+	resp := make(map[string]string)
+
+	c.RequireUserId()
+	if c.Err != nil {
+		return
+	}
+
+	if statusMap, err := app.GetUserStatusesByIds([]string{c.Params.UserId}); err != nil {
+		c.Err = err
+		return
+	} else {
+		resp["status"] = statusMap[0].Status
```

Comment:
I think we need to check if `len(statusMap) == 0` first and return a 404 NotFound error if it is. Also can we use `statusMap[0].ToJson()` as the response body?